lodash.js

```diff
@@ -3450,6 +3450,27 @@
     }
 
     /**
+     * The base implementation of `_.update`.
+     *
+     * @private
+     * @param {Object} object The object to query.
+     * @param {Array|string} path The path of the property to update.
+     * @param {Function} func The updater function.
+     * @param {Array} args The provided extra arguments for updater `func`.
+     * @returns {Object} Returns `object`.
+     */
+    function baseUpdate(object, path, func, args) {
+      if (object == null) {
+        return null;
+      }
+      var value = baseGet(object, path),
+          updated = isFunction(func)
+            ? func.apply(object, [value].concat(args))
```

Comment:
The `update` function should use `baseCastFunction` so that `baseUpdate` will always get a function and the `isFunction` check can be removed. Since there's no args the `func` invocation can be simplified.